src/envoy/http/authn/http_filter_factory.cc

```diff
@@ -74,12 +80,48 @@ class AuthnFilterConfig : public NamedHttpFilterConfigFactory,
   std::string name() override { return kAuthnFactoryName; }
 
  private:
-  HttpFilterFactoryCb createFilter() {
+  // Convert istio-authn::jwt to jwt_auth::jwt in protobuf format.
+  void convertJwtAuthFormat(
+      istio::authentication::v1alpha1::Policy& policy,
+      Http::JwtAuth::Config::AuthFilterConfig* proto_config) {
+    int origins_size = 0;
+    // In POC, only inspect the first credential_rule
+    if (policy.credential_rules_size() > 0 &&
+        policy.credential_rules()[0].origins_size() > 0) {
+      origins_size = policy.credential_rules()[0].origins_size();
+    }
+    if (origins_size > 0) {
+      const ::istio::authentication::v1alpha1::OriginAuthenticationMethod& m =
+          policy.credential_rules()[0].origins()[0];
+      if (m.has_jwt()) {
+        // In POC, only the following fields are converted.
+        // Todo: may need to convert more fields if necessary
+        Http::JwtAuth::Config::JWT jwt;
+        jwt.set_issuer(m.jwt().issuer());
+        jwt.set_jwks_uri(m.jwt().jwks_uri());
+        jwt.set_jwks_uri_envoy_cluster(kJwtClusterName);
+        auto jwts = proto_config->add_jwts();
```

Comment:
I think two "JWT" proto are very close,  you can use Envoy  MessageUtil::jsonConvert to convert them

MessageUtil::jsonConvert(m.jwt(),  *proto_config->add_jwts());

Reply:
Done.

Reply:
As a mater of fact, I'm not sure if those protos will be kept aligned in the future. Doing manual re-construct the proto may be a safer in the long run.

Reply:
When these protos diverge, we can change to manually re-construct. A to-do item has been added to the conversion function.